### ISSUE TITLE
tslint is going crazy when installed via yarn 1.1.0

### ISSUE THREAD

**Issue by Hotell** [palantir/tslint#3277]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__:   2.2.2
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
function createSpacebarKeypressEvent() {
    return new KeyboardEvent(
      'keypress',
      {
        keyCode: 32,
        which: 32,
        charCode: 32,
      } as any
    )
  }
```

![image](https://user-images.githubusercontent.com/1223799/31172791-44bb354c-a905-11e7-96a7-49622ba2ce5b.png)

havoc follows:

![image](https://user-images.githubusercontent.com/1223799/31172874-90ef86ca-a905-11e7-806a-e393b0cf087a.png)

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest"
  ],
  "jsRules": {},
  "rulesDirectory": [],
  "rules": {}
}
```

#### Actual behavior

when we switched to latest yarn 1.1.0 all builds started to fail on described tslint super weird errors

with `npm install` everything works as expected

#### Expected behavior

no errors

**Comment by adidahiya** [comment]
I'd prefer to keep the discussion to #3251; do you think this is materially different? The fix will likely be the same for both issues.

**Comment by Hotell** [comment]
fair enough, although #3251 has kinda confusing title ... :)
